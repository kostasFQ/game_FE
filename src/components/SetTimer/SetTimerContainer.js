import { connect } from 'react-redux';
import SetTimer from './SetTimerComponent';
import { setGameTimer } from 'store/game/actions';
import seconds from 'helpers/secondsArray';

const mapDispatchToProps = dispatch => ({
  setGameTimer: seconds => dispatch(setGameTimer(seconds)),
})

const mapStateToProps = state => ({
  game: state.game,
  seconds
});

export default connect(mapStateToProps, mapDispatchToProps)(SetTimer);