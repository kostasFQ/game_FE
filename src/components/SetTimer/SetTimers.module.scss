.setTimers {
  &__container {
    
    &__checkboxes {
      display: flex;
      height: 70px;
      
      &__item {
        width: 20%;
        cursor: pointer;
        outline: none;
        border: none;
        transition: 0.5s;
        font-family: 'Saira', sans-serif;
        font-size: 2em;
        background: none;
        
        &_active {
          color: red;
          font-size: 3em;
        }
      }
    }
  }
}