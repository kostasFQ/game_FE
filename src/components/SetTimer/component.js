import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import cn from 'classnames';
import styles from './SetTimers.module.scss';

const seconds = [5, 10, 15, 30, 60];

class SetTimer extends Component {

  select= e => {
    const { target: { value } } = e;
    const { setGameTimer } = this.props;
    setGameTimer(value);
  }

  render() {
    const { game: { gameStarted, gameOver, initialTime } } = this.props;
    
    return (
      <form onClick={this.select} className={styles.setTimers__container}>
        <div>select how many seconds you want</div>
        <div className={styles.setTimers__container__checkboxes}>
          {
            seconds.map(i => (
              <input name='seconds' type='button' value={i} key={i} disabled={gameStarted && !gameOver}
                className={cn(
                  styles.setTimers__container__checkboxes__item,
                  { [styles.setTimers__container__checkboxes__item_active]: i === +initialTime },
                )}
              />) ) 
          }
        </div>
      </form>
    );
  }
}

export default withRouter(SetTimer);