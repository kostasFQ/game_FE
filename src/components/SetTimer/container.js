import { connect } from 'react-redux';
import SetTimer from './component';
import { setGameTimer } from 'store/game/actions';

const mapDispatchToProps = dispatch => ({
  setGameTimer: seconds => dispatch(setGameTimer(seconds)),
})

const mapStateToProps = state => ({
  game: state.game,
});

export default connect(mapStateToProps, mapDispatchToProps)(SetTimer);