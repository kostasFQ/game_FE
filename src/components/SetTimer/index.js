import SetTimer from './SetTimerContainer';

export default SetTimer;