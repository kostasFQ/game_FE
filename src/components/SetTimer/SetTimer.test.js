import React from 'react';
import renderer from 'react-test-renderer';
import _store from '../../../mocks/store';
import SetTimer from './component';

const props = {
  game: {
    gameStarted: true,
    gameOver: true,
    initialTime: 10
  },
  seconds: [5, 10, 15, 30, 60],
  setGameTimer: jest.fn( () => {console.log('submit')} ) 
}

const instance = renderer.create(
  <SetTimer {...props} />
);

test('FinalCount should be render', () => {
  let component = instance.toJSON();
  expect(component).toMatchSnapshot();
})

test('Form submit', () => {
  const x = instance.root;
  const form = x.findByType('form')
  const e ={ target: { value: 10 } }
  form.props.onClick(e);
  expect(props.setGameTimer).toHaveBeenCalled()
})