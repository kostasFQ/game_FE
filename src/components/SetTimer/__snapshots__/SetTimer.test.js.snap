// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinalCount should be render 1`] = `
Array [
  <div>
    select how many seconds you want
  </div>,
  <form
    className="setTimers__container"
    onClick={[Function]}
  >
    <div
      className="setTimers__container__checkboxes"
    >
      <input
        className="setTimers__container__checkboxes__item"
        disabled={false}
        name="seconds"
        type="button"
        value={5}
      />
      <input
        className="setTimers__container__checkboxes__item"
        disabled={false}
        name="seconds"
        type="button"
        value={10}
      />
      <input
        className="setTimers__container__checkboxes__item setTimers__container__checkboxes__item_active"
        disabled={true}
        name="seconds"
        type="button"
        value={15}
      />
      <input
        className="setTimers__container__checkboxes__item"
        disabled={false}
        name="seconds"
        type="button"
        value={30}
      />
      <input
        className="setTimers__container__checkboxes__item"
        disabled={false}
        name="seconds"
        type="button"
        value={60}
      />
    </div>
  </form>,
]
`;
